src/java/org/apache/cassandra/service/StorageService.java

```diff
@@ -1855,14 +1855,21 @@ public Collection<InetAddressAndPort> prepareForBootstrap(long schemaTimeoutMill
                 }
 
                 // check for operator errors...
+                long nanoDelay = MILLISECONDS.toNanos(ringTimeoutMillis);
```

Comment:
no need to recalculate this value in every iteration of the loop. So moved outside of the for loop

Reply:
also using `MILLISECONDS.toNanos` here to avoid conversion errors